app/models/download.rb

```diff
@@ -40,16 +40,17 @@ def set_file_type(options)
 
   # Given a mime type, return a file extension
   def file_type_from_mime(mime)
-    ext = MimeMagic.new(mime.to_s).subtype
-    case ext
+    subtype = Marcel::MimeType.for(mime).to_s
```

Comment:
It looks like `Marcel::MimeType`'s `for` structures its arguments differently than `MimeMagic#new`: https://github.com/rails/marcel/blob/12fc8daae656ccb09441c4a3c376b5de3af05172/lib/marcel/mime_type.rb#L29

You can try the `Marcel::Magic` class instead:
```suggestion
    subtype = Marcel::Magic.new(mime.to_s).subtype
```